src/LDLFactorizations.jl

```
@@ -130,11 +287,21 @@ mutable struct LDLFactorization{T<:Real,Ti<:Integer}
 end
 
 # use AMD permutation by default
-ldl(A::Array{T,2}, args...) where T<:Real = ldl(sparse(A), args...)
-ldl(A::SparseMatrixCSC{T,Ti}) where {T<:Real,Ti<:Integer} = ldl(A, amd(A))
+ldl(A::Array{T,2}, args...; upper = false) where T<:Real = ldl(sparse(A), args...; upper = upper)
+ldl(A::SparseMatrixCSC{T,Ti}; upper = false ) where {T<:Real,Ti<:Integer} = ldl(A, amd(A); upper = upper)
+
+# symmetric matrix input
+ldl(sA::Symmetric{T,SparseMatrixCSC{T,Ti}}) where {T<:Real,Ti<:Integer} =
+  if sA.uplo == 'L'
+    A = sparse(sA.data')
+    ldl(A, amd(A); upper = true )
```

Here, we should return an error instead of generating a new sparse matrix.